pkl-cli/pkl-cli.gradle.kts

```diff
@@ -210,8 +215,14 @@ fun Exec.configureExecutable(
         // executable
         if (!buildInfo.isReleaseBuild) {
           add("-Ob")
+        } else {
+          add("-Os")
+        }
+        if (buildInfo.isNativeArch) {
+          add("-march=native")
+        } else {
+          add("-march=compatibility")
         }
-        add("-march=compatibility")
```

Comment:
changes here:

- when building a release...
  - `-Os` is passed, which optimizes for size (this is a new feature supported by `native-image`)
- when building with `-DnativeArch=true`...
  - `-march=native` is passed
  - otherwise, `-march=compatibility` is passed (no change)
- when not building a release...
  - `-Ob` is passed to optimize build time (no change)